packages/platforms/browser/lib/auto-instrumentation/network-request-plugin.ts

```diff
@@ -59,15 +69,49 @@ export class NetworkRequestPlugin implements Plugin<BrowserConfiguration> {
     span.setAttribute('http.method', startContext.method)
     span.setAttribute('http.url', networkRequestInfo.url)
 
-    return (endContext: RequestEndContext) => {
-      if (endContext.state === 'success') {
-        span.setAttribute('http.status_code', endContext.status)
-        this.spanFactory.endSpan(span, endContext.endTime)
-      }
+    return {
+      onRequestEnd: (endContext: RequestEndContext) => {
+        if (endContext.state === 'success') {
+          span.setAttribute('http.status_code', endContext.status)
```

Comment:
Couldn't `span` be null now?

Reply:
no

Reply:
Ah yes I see - the span null path in `getExtraRequestHeaders` is from the call on line 54.